helpers/keyFromSlug.ts

```diff
@@ -1,8 +1,14 @@
 import router from "next/router";
-
 import idFromSlug from "./idFromSlug";
+import isServer from "./isServer";
+
+const keyFromSlug = (): (() => string) => (): string => {
+  if (isServer) {
+    return "";
+  }
 
-const keyFromSlug = (): (() => string) => (): string =>
-  process.browser ? idFromSlug(router.query.slug) : "";
+  // `router.query` might be empty when first loading a page for some reason.
```

Comment:
does this mean we'll still get a console error here? If so, is it worth adding a guard clause for this instead, or would that complicate things a lot?

Reply:
No, it's more an explanation of why we guard for something that _should_ always be present. What I mean by this comment is that `router.query == {}`.